src/shim/state_tree.rs

```diff
@@ -291,6 +291,36 @@ where
             StateTree::V0(_) => bail!("StateTree::set_actor not supported on old state trees"),
         }
     }
+
+    /// Returns the public key type of
+    /// address(`BLS`/`SECP256K1`) of an actor identified by `addr`,
+    /// or its delegated address.
+    pub fn resolve_to_deterministic_addr(
+        &self,
+        store: &impl Blockstore,
+        addr: Address,
+    ) -> anyhow::Result<Address> {
+        use crate::shim::address::Protocol::*;
+        match addr.protocol() {
+            BLS | Secp256k1 | Delegated => Ok(addr),
+            _ => {
+                let actor = self
+                    .get_actor(&addr)?
+                    .with_context(|| format!("failed to find actor: {addr}"))?;
+
+                // if state.Version() >= types.StateTreeVersion5
```

Comment:
```suggestion
```

Reply:
This is the Go code. Rust api does not expose state tree version so I have to use some workaround here

Reply:
So let's better have a comment than a commented-out Go code. :)